Automattic/wp-calypso
# Write Button: Clarify through styling that the number of posts shown to the right is a menu button

The Write button in the master bar has a number in an oval if there are drafts to complete.
<img width="252" alt="Screenshot 2021-08-19 at 8 52 51 PM" src="https://user-images.githubusercontent.com/1500769/130040077-c54eda74-9665-4b39-a12d-1b802d01cfce.png">

Clicking the number opens a popup.
<img width="407" alt="Screenshot 2021-08-19 at 8 58 21 PM" src="https://user-images.githubusercontent.com/1500769/130040244-2b1cdf49-c40f-4188-9fd4-cfdcf3a8be4c.png">

#51834 pointed out the button didn't have great affordance.

#### Changes proposed in this Pull Request

* Update style to match design h/t @ollierozdarz p1629086395035300-slack-C029SB8JT8S
* Add chevron to the draft section of the button so it's clear a popup will appear
* Update icon
* Hide the number of drafts on small screens when there are more than 2 digits

No drafts:
<img width="252" alt="Screenshot 2021-08-19 at 8 53 15 PM" src="https://user-images.githubusercontent.com/1500769/130040456-b262f21f-3117-487c-ba65-9d18093daea5.png">

One draft:
<img width="252" alt="Screenshot 2021-08-19 at 8 51 03 PM" src="https://user-images.githubusercontent.com/1500769/130040407-0d73f12e-3e3d-4ec1-afa8-c7e1d626fd59.png">

Many drafts:
<img width="252" alt="Screenshot 2021-08-19 at 8 51 34 PM" src="https://user-images.githubusercontent.com/1500769/130040425-95549045-82ec-46fe-bc75-e47648a8332e.png">

A different colour scheme:
<img width="252" alt="Screenshot 2021-08-19 at 8 52 10 PM" src="https://user-images.githubusercontent.com/1500769/130040437-3dd44a22-bd46-4a38-b545-81da1e6cd8a8.png">

A whole bunch of mobile variations. Notice that the number of drafts is hidden on small screens when there are more than 2 digits:
<img width="1216" alt="Screenshot 2021-08-20 at 3 18 33 PM" src="https://user-images.githubusercontent.com/1500769/130174335-05e893fc-e5d9-4b61-a8f5-ec7b90bde7c1.png">


#### Testing instructions



* I've tried to chunk the commits into reviewable bits
* Calypso only change, can be tested using `calypso.live`
* Test with no drafts
* Test with a number of drafts
* Test mobile layout
* Test using `?disable-nav-unification` to make sure the non-unified UI isn't completely broken. All users are seeing the unifified UI, but we've had a user setting in the past that toggled between the two. We probably don't want to break it in case the user toggle comes back.



Related to #51834
